api/job/dispatcher.go

```diff
@@ -0,0 +1,62 @@
+package job
+
+import (
+	"log"
+	"time"
+
+	"github.com/quintilesims/layer0/common/models"
+)
+
+const (
+	DISPATCHER_PERIOD = time.Second * 5
+)
+
+func RunWorkersAndDispatcher(numWorkers int, store Store, runner Runner) *time.Ticker {
+	queue := make(chan models.Job)
+	for i := 0; i < numWorkers; i++ {
+		worker := NewWorker(i+1, store, queue, runner)
+		worker.Start()
+	}
+
+	dispatcher := NewDispatcher(store, queue)
+	ticker := time.NewTicker(DISPATCHER_PERIOD)
+	go func() {
+		for range ticker.C {
```

Comment:
Oh woah, you can `range` a channel? That's super cool. So with something like a Ticker channel, you're just fetching the next item in a (controllably infinite) list, where items come out at specific intervals?

Sort of similar to something like
```
while True:
    sleep 5
    do_thing()
```
only you can control it with `Ticker.Stop()`, and it's smarter about keeping accurate time than a sleep would be.

Reply:
Correct, it isn't greedy since the underlying mechanism in a channel